addons/web_editor/static/src/js/editor/odoo-editor/test/spec/copyPaste.test.js

```diff
@@ -122,6 +123,17 @@ describe('Cut', () => {
                 contentAfter: '<p>[]<br></p>',
             });
         });
+        it('should restore content and selection when cut followed by undo', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: '<p>a[bcd]e</p>',
+                stepFunction: async editor => {
+                    const clipboardData = new DataTransfer();
+                    triggerEvent(editor.editable, 'cut', { clipboardData });
+                    undo(editor);
+                },
+                contentAfter: '<p>a[bcd]e</p>',
```

Comment:
I'm confused as to why would this test break if the fix wasn't present. What would be the result of this test without the corresponding fix ? Because bcd would be back anyway, right ? So would it just be the selection that would be wrong ? I think the test should also include other actions. For example a deleteRange, which would rollback the cut alongside it, in a similar fashion as this bug report: https://discord.com/channels/678381219515465750/687338152372994236/1071004267298697267

Reply:
I have just pushed an updated test to explicitly check the expected registration of a history step.
And I've added a new one to test the referred bug report.